### ISSUE TITLE
SSID isn't broadcast on boot

### ISSUE THREAD

**Issue by Ph-ill** [loiccoyle/tinyticker#34]
Hey, I'm not sure if I have an incompatible board or something but whenever I attempt to flash an sd and boot it in a pi zero w 1.1 I'm unable to find a broadcasted wifi network from the device, nor am I able to connect to it via a usb ethernet adapter. 

It also doesn't write anything to the display, I'm using a waveshare 2.13 v3. 

Weirdly a blank comitup image works but then after installing tinyticker the services always exit out with an error code and nothing is written to the screen.

**Comment by Ph-ill** [comment]
After viewing the boot output on a monitor I can now confirm there is some issue with the comitup service failing to start.

**Comment by Ph-ill** [comment]
Looks like it was a problem with the NetworkManager service running 

`sudo systemctl enable NetworkManager.service' 

and rebooting solved the issue. 

I'm not going to close this because I still have have nothing being outputted to the screen.

**Comment by Ph-ill** [comment]
The tinyticker services weren't in the correct folder

running

`cd /lib/systemd/system/
sudo wget https://github.com/loiccoyle/tinyticker/raw/main/systemd/tinyticker-qrcode.service
sudo wget https://github.com/loiccoyle/tinyticker/raw/main/systemd/tinyticker-web.service
sudo wget https://github.com/loiccoyle/tinyticker/raw/main/systemd/tinyticker.service
sudo systemctl enable tinyticker-qrcode.service
sudo systemctl enable tinyticker-web.service
sudo systemctl enable tinyticker.service
`
resolved this, however the services were failing to start so they had to be modified as per the PR i've made here #35 

There now seems to be an issue with yfinance not being able to pull the data correctly, updating to the latest version with

'pip install yfinance==0.2.33'

did not resolve the issue but is likely required as it includes a number of fixes resolving issues regarding yahoo finance changes on their backend. 

I'm recieving 
'15:15:33 - tinyticker.ticker - DEBUG: Ticker    stock   GME     16x15m  900 response outdated, skipping.'
in the tinyticker logs and seemingly nothing is appearing on the display as a result.

**Comment by loiccoyle** [comment]
Sorry for not addressing this earlier, I will try to figure it out this week.

> changing skip_outdated: bool = True, to skip_outdated: bool = False, in ticker.py allows data to printed to the screen at least but I'm guessing it's not the most recent data.

Were you testing during open hours of the appropriate stock exchange?

**Comment by Ph-ill** [comment]
No worries at all, I appreciate the response, would still love to get this working and contribute to the project myself as it's by far the most user friendly rpi zero stock ticker. 

I was testing with TSLA on the NASDAQ at about 4pm GMT so all my testing should have been within open hours. Interestingly with no change on my part it stopped displaying a chart and started printing code to the screen as shown in the picture below. 

![IMG-20231218-WA0008](https://github.com/loiccoyle/tinyticker/assets/3920056/6d353d1a-e1e0-4dd3-9daf-3c3747b59caf)

This stumped me at the time so I've not looked back into it since I'll see if I can gather some more info for you tomorrow.

**Comment by loiccoyle** [comment]
hmm it looks like it is displaying a partial crash log, would be good to see the full log.

On my end, I spent some time to update the tinyticker image hoping it would solve your issues, but I'm having issues getting anything to display at all on my rpi, even with the waveshare examples... so it is taking me longer than I expected to address this. Will keep trying and keep you posted.

**Comment by loiccoyle** [comment]
Ok I got it working again on my end. I updated the image on the drive, https://drive.google.com/drive/folders/1U-PGzkOtSynN6FGDq2MsXF9kXGdkzd0D?usp=share_link can you try it?

**Comment by Ph-ill** [comment]
Got it working but it required running


`cd /lib/systemd/system/`
`sudo wget https://github.com/loiccoyle/tinyticker/raw/main/systemd/tinyticker-qrcode.service`
`sudo wget https://github.com/loiccoyle/tinyticker/raw/main/systemd/tinyticker-web.service`
`sudo wget https://github.com/loiccoyle/tinyticker/raw/main/systemd/tinyticker.service`
`sudo systemctl enable tinyticker-qrcode.service`
`sudo systemctl enable tinyticker-web.service`
`sudo systemctl enable tinyticker.service`

and adding `User=tinyticker` to the service files and reloading them as mentioned previously, all of the other issues seem to be resolved however! 

I need to reset comitup and package up this pi as it was supposed to be a christmas gift for someone but tomorrow I'll take a look at the pi-gen repo and see if I can figure out why the service files aren't ending up in the image. 

I appreciate the help.

**Comment by Ph-ill** [comment]
Oh I should probably mention that I did also try enable the service files as the tinyticker user but still ran into the same issue as before, the only thing that's worked is editing the service files and enabling them as root

**Comment by loiccoyle** [comment]
> and adding User=tinyticker to the service files and reloading them as mentioned previously, all of the other issues seem to be resolved however!

I'm glad the other issues were solved.

> tomorrow I'll take a look at the pi-gen repo and see if I can figure out why the service files aren't ending up in the image.

The files are ending up in the image, they are configured as [user services](https://wiki.archlinux.org/title/Systemd/User) and are placed in the `/home/tinyticker/.config/systemd/user/` directory.

I think I solved the issue, see https://github.com/loiccoyle/pi-gen/commit/c11bb699e387850308f1f067cbbf46485a4a588d from what I can tell it was caused by the service file not being able to connect the `STDOUT` of the process to the log file during the `ExecStartPre` command which was used to create the log dir.

So instead I move the [log directory creation to the image building step](https://github.com/loiccoyle/pi-gen/blob/master/stage2/05-tinyticker/01-run.sh#L28) which is cleaner anyway. I am building the image one last time and will do a final test.
